mbed-client-libservice/ip6string.h

```diff
@@ -69,6 +69,19 @@ void stoip6(const char *ip6addr, size_t len, void *dest);
  */
 unsigned char sipv6_prefixlength(const char *ip6addr);
 
+/**
+ * Convert numeric IPv6 address string with prefix to a binary.
+ *
+ * IPv4 tunneling addresses are not covered.
+ *
+ * \param ip6addr IPv6 address in string format.
+ * \param dest buffer for address. MUST be 16 bytes.
+ * \param prefix_len_out length of prefix
```

Comment:
Document that this is filled as -1 if no prefix length given.

I note you permit this to be NULL. If it is NULL, is it best to just process and ignore the prefix length? Or could stoip6_prefix(addr, dest, NULL) actually be "just process address", so it faults if there is a prefix, and saves the external strlen?

Reply:
Assuming that processing and ignoring the prefix_length is more flexible for the application.